pkg/ext/delegate_error_test.go

```diff
@@ -0,0 +1,1109 @@
+
+			},
+			wantErr: true,
+		},
+		{
+			name: "get worked - update validation error",
+			input: input{
+				parentCtx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				name: "test-user",
+				createValidation: func(ctx context.Context, obj runtime.Object) error {
+					return nil
+				},
+				updateValidation: func(ctx context.Context, obj, old runtime.Object) error {
+					return assert.AnError
+				},
+			},
+			expect: output{
+				obj:     nil,
+				created: false,
+				err:     assert.AnError,
+			},
+			setup: func(objInfo *MockUpdatedObjectInfo, store *MockStore[*TestType, *TestTypeList]) {
+				store.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(&TestType{}, nil)
+				objInfo.EXPECT().UpdatedObject(gomock.Any(), gomock.Any()).Return(&TestType{}, nil)
+
+			},
+			wantErr: true,
+		},
+		{
+			name: "get worked - store update error",
+			input: input{
+				parentCtx: request.WithUser(context.Background(), &user.DefaultInfo{
+					Name: "test-user",
+				}),
+				name: "test-user",
+				createValidation: func(ctx context.Context, obj runtime.Object) error {
+					return nil
+				},
+			},
+			expect: output{
+				obj:     nil,
+				created: false,
+				err:     assert.AnError,
+			},
+			setup: func(objInfo *MockUpdatedObjectInfo, store *MockStore[*TestType, *TestTypeList]) {
+				store.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(&TestType{}, nil)
+				objInfo.EXPECT().UpdatedObject(gomock.Any(), gomock.Any()).Return(&TestType{}, nil)
+				store.EXPECT().Update(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, assert.AnError)
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			addToSchemeTest(scheme)
+
+			gvk := schema.GroupVersionKind{Group: "ext.cattle.io", Version: "v1", Kind: "TestType"}
+			gvr := schema.GroupVersionResource{Group: "ext.cattle.io", Version: "v1", Resource: "testtypes"}
+
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := NewMockStore[*TestType, *TestTypeList](ctrl)
+			mockObjInfo := NewMockUpdatedObjectInfo(ctrl)
+			tt.input.objInfo = mockObjInfo
+			tt.setup(mockObjInfo, mockStore)
+
+			testDelegate := &delegateError[*TestType, *TestTypeList]{
+				inner: &delegate[*TestType, *TestTypeList]{
+					scheme: scheme,
+					t:      &TestType{},
+					tList:  &TestTypeList{},
+					gvk:    gvk,
+					gvr:    gvr,
+					store:  mockStore,
+				},
+			}
+
+			obj, created, err := testDelegate.Update(tt.input.parentCtx, tt.input.name, tt.input.objInfo, tt.input.createValidation, tt.input.updateValidation, tt.input.forceAllowCreate, tt.input.options)
+
+			if tt.wantErr {
+				// check if we have an error
+				assert.Error(t, err)
+
+				// check if error is an apierror
+				if _, ok := err.(apierrors.APIStatus); ok {
+					assert.True(t, ok)
+					return
+				}
```

Comment:
Should this be

```suggestion
				_, ok := err.(apierrors.APIStatus)
				assert.True(t, ok)
```

? Right now if the error returned is not an APIStatus there just won't be any assert.